QGL/PulsePrimitives.py

```diff
@@ -716,14 +716,18 @@ def CNOT_simple(source, target, **kwargs):
     p = X(channel, **kwargs)
     return p._replace(label="CNOT")
 
-try:
-    cnot_impl = globals()[config.cnot_implementation]
-except:
-    raise NameError("A valid CNOT implementation was not defined [{}]".format(config.cnot_implementation))
 
 @_memoize
 def CNOT(source, target, **kwargs):
-    return cnot_impl(source, target, **kwargs)
+    channel = ChannelLibraries.EdgeFactory(source, target)
+    if not channel.cnot_impl:
+        print('Warning: CNOT implementation undefined for {}'.format(channel.label))
```

Comment:
Here we assume CR, but don't tell the user that's what we've chosen for them? I'm not sure what the best strategy is.

Reply:
I agree, we should at least print what the default choice is. My guess is to leave CR as the default, being the most common choice for now